### PR TITLE
Debug name, linkto and edit person

### PR DESCRIPTION
**Pull request by yiwen101**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Name.java
Comment (pzl111) [inline src/main/java/seedu/address/model/person/Name.java:1]: what was the purpose of changing it to case insensitive again? i forgot

**Comment by yiwen101** [comment]
Fix #226

**Comment by yiwen101** [comment]
Fix #192
Fix #189 
Fix #225 
Fix #221 
Fix #179 
Fix #158 
Fix #177
Fix #212

File: src/main/java/seedu/address/logic/commands/AddLessonCommand.java [hunk 1]
```diff
@@ -26,6 +26,9 @@ public CommandResult execute(Model model) throws CommandException {
 
         if (model.hasLessonClashWith(lesson)) {
             Lesson clashingLesson = model.getLessonClashWith(lesson);
+            if (clashingLesson.getName().equals(lesson.getName())) {
```
Comment (HollaG) [inline src/main/java/seedu/address/logic/commands/AddLessonCommand.java:29]: Is this only for lessons which have the same time? 

I forgot if we decided on unique lesson names (which i think may be classified as a bug - what if a tutor wants to have the same lesson on two diff dates with differnet people?)

**Review by HollaG — APPROVED** [review]
LGTM just one minor comment

**Pull request merged**